modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/QuantumToolkit.java

```diff
@@ -1478,6 +1478,62 @@ private void draw(Graphics g, int x, int y, int w, int h) {
 
             }
 
+            private void renderTile(int x, int xOffset, int y, int yOffset, int w, int h,
+                                    IntBuffer buffer, ResourceFactory rf, QuantumImage tileImg, QuantumImage targetImg) {
+                com.sun.prism.RTTexture rt = tileImg.getRT(w, h, rf);
+                if (rt == null) {
+                    return;
+                }
+                Graphics g = rt.createGraphics();
+                draw(g, x + xOffset, y + yOffset, w, h);
+                int[] pixels = rt.getPixels();
+                if (pixels != null) {
+                    buffer.put(pixels);
+                }
+                else {
+                    rt.readPixels(buffer, rt.getContentX(), rt.getContentY(), w, h);
+                }
+                //Copy tile's pixel into the target image
+                targetImg.image.setPixels(xOffset, yOffset, w, h,
+                        javafx.scene.image.PixelFormat.getIntArgbPreInstance(), buffer, w);
+                rt.unlock();
+            }
+
+            private void renderWholeImage(int x, int y, int w, int h, ResourceFactory rf, QuantumImage pImage) {
+                com.sun.prism.RTTexture rt = pImage.getRT(w, h, rf);
+                if (rt == null) {
+                    return;
+                }
+                Graphics g = rt.createGraphics();
+                draw(g, x, y, w, h);
+                int[] pixels = rt.getPixels();
+                if (pixels != null) {
+                    pImage.setImage(com.sun.prism.Image.fromIntArgbPreData(pixels, w, h));
+                }
+                else {
+                    IntBuffer ib = IntBuffer.allocate(w * h);
+                    if (rt.readPixels(ib, rt.getContentX(), rt.getContentY(), w, h)) {
+                        pImage.setImage(com.sun.prism.Image.fromIntArgbPreData(ib, w, h));
+                    }
+                    else {
+                        pImage.dispose();
+                        pImage = null;
+                    }
+                }
+                rt.unlock();
+            }
+
+            private int computeOptimumTileSize(int size, int maxSize, AtomicBoolean isDivExact) {
+                for (int n = 2; n <= 6; n++) {
```

Comment:
It would be helpful if you add explanation about the constants 2 and 6 here.

Reply:
Good point.
I've added the following comment:

>// This methods attempts to find the smallest integer divider for the provided `size`
>// while the result of the division is less than `maxSize`.
>// It tests all potential dividers from 2 to 6 and returns the result of the division
>// if all conditions can be satisfied or, failing that, `maxSize`.
>// The value for `isDivExact` reflects whether or not an exact divider could be found.

Please note that my choice of 6 for the largest attempted divider is totally arbitrary; we ideally want to maximize the chances to find an exact divider, but we also need to avoid making the tiles too small in the process and it seemed a good compromise.